src/System.Data.SqlClient/tests/ManualTests/SQL/AsyncTest/XmlReaderAsyncTest.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using Xunit;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public static class XmlReaderAsyncTest
+    {
+        private static string commandText =
+            "SELECT * from dbo.Customers FOR XML AUTO, XMLDATA;";
+
+        [CheckConnStrSetupFact]
+        public static void ExecuteTest()
+        {
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TcpConnStr))
+            {
+                try
+                {
+                    SqlCommand command = new SqlCommand(commandText, connection);
+                    connection.Open();
+
+                    IAsyncResult result = command.BeginExecuteXmlReader();
+                    while (!result.IsCompleted)
+                    {
+                        System.Threading.Thread.Sleep(100);
+                    }
+
+                    XmlReader reader = command.EndExecuteXmlReader(result);
+
+                    reader.ReadToDescendant("dbo.Customers");
+                    Assert.Equal("ALFKI", reader["CustomerID"]);
+                }
+                catch (SqlException ex)
+                {
+                    Console.WriteLine("Error ({0}): {1}", ex.Number, ex.Message);
```

Comment:
Seems you can eliminate all the try/catches . If the code throws, the test will fail anyway, and do ToString on the exception, which will show all the same information (and more)

Reply:
Agree on both of the above points and have updated the tests.